packages/pwa/app/utils/locale.js

```diff
@@ -53,11 +34,16 @@ export const loadLocaleData = async (locale) => {
  * Get all of the locale-related configuration data
  * @param {Object} options
  * @param {function} [options.getUserPreferredLocales] - Identify what set of locales the user prefers
+ * @param {object} [options.l10nConfig] - l10n configuration object
  * @returns {Promise<Object>} The configuration data
  */
-export const getLocaleConfig = async ({getUserPreferredLocales} = {}) => {
-    const preferredLocales = getUserPreferredLocales ? getUserPreferredLocales() : [DEFAULT_LOCALE]
-    const targetLocale = whichLocaleToLoad(preferredLocales, supportedLocales, DEFAULT_LOCALE)
+export const getLocaleConfig = async ({getUserPreferredLocales, l10nConfig = {}} = {}) => {
```

Comment:
I like the idea of changing the name. This function does and returns a lot of things. Is that completely captured in findAndLoadTargetLocale ?, I’m not completely sold.
If I was going to think of a new name for this function, I’d look at the data being returns and what it’s used for. Because there is so much data being returned, on top of loading messages in a async manner, it’s difficult to use it all in the function name, it would end up like getTargetSupportedAndDefaultLocalesWithMessages :joy: So instead I look and see that the data is being used mainly for the IntlProvider component. Could we simply called it something like getIntlProviderProps?